Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -2471,8 +2489,14 @@ void InlineCacheCompiler::emitProxyObjectAccess(ProxyObjectAccessCase& accessCas
 
     done.link(&jit);
 
-    int stackPointerOffset = (codeBlock->stackPointerOffset() * sizeof(Register)) - m_preservedReusedRegisterState.numberOfBytesPreserved - spillState.numberOfStackBytesUsedForRegisterPreservation;
-    jit.addPtr(CCallHelpers::TrustedImm32(stackPointerOffset), GPRInfo::callFrameRegister, CCallHelpers::stackPointerRegister);
+    if (codeBlock->useDataIC()) {
+        jit.loadPtr(CCallHelpers::Address(GPRInfo::constantsRegister, BaselineJITData::offsetOfStackOffset()), m_scratchGPR);
```

Comment:
We should really create an alias for `GPRInfo::constantsRegister` like `GPRInfo::unlinkedCallJITDataRegister` or something. It's pretty confusing right now.